src/utils/pureComments.ts

```diff
@@ -0,0 +1,54 @@
+// @ts-ignore
+import walk from 'acorn/dist/walk';
+import * as ESTree from 'estree';
+import { CommentDescription } from '../Module';
+
+type ESTreeNodeWithLocation = ESTree.Node & { start: number; end: number };
+type NodeHandler = (node: ESTree.Node) => void;
+
+function checkCommentsBeforeNode(
+	node: ESTreeNodeWithLocation,
+	st: { handleNode: NodeHandler; commentIdx: number; commentNodes: CommentDescription[] }
```

Comment:
I think these types are correct, but I don't really know. I guess you'll have to look at them in your IDE? I'll get mine set up, but I don't have time right now.